openwisp_monitoring/device/apps.py

```diff
@@ -30,3 +51,26 @@ def device_post_delete_receiver(cls, instance, **kwargs):
         instance.__class__ = DeviceData
         instance.checks.all().delete()
         instance.metrics.all().delete()
+
+    @classmethod
+    def _generate_cache_key(cls, device):
```

Comment:
you may use a simple function and store it here: https://github.com/openwisp/openwisp-monitoring/blob/master/openwisp_monitoring/device/utils.py

I think it could be better since you'll be using that in tests as well. It makes sense as it's not something specifically related to the AppConfig class.

I think it can be named simply `get_device_recovery_cache_key'